### PR TITLE
Ruby 3 upgrade

### PR DESCRIPTION
**Pull request by jjoyner11**

### CODE REVIEW

File: lib/fdoc/spec_watcher.rb [hunk 1]
```diff
@@ -8,7 +8,8 @@ module SpecWatcher
       define_method(verb) do |*params|
         action, request_params = params
 
-        super(*params)
+        request_params.nil? ? 
+        super(*params) : super(action, **request_params)
```
Comment (maxfierke) [inline lib/fdoc/spec_watcher.rb]: one line is fine here for the ternary

```suggestion
        request_params.nil? ? super(*params) : super(action, **request_params)
```

I think we could also use [`ruby2_keywords`](https://github.com/ruby/ruby2_keywords) here by calling `ruby2_keywords verb` after the `define_method` (if `ruby2_keywords` is defined), but this also seems fine

File: spec/fdoc/endpoint_scaffold_spec.rb [hunk 2]
```diff
@@ -113,7 +113,7 @@
     it "produces a valid JSON schema for the response" do
       subject.consume_request(request_params)
       subject.request_parameters["properties"].should have(3).keys
-      JSON::Validator.validate!(subject.request_parameters, request_params, :version => :draft3).should be_true
+      JSON::Validator.validate!(subject.request_parameters, request_params, :version => :draft3).should be_truthy
```
Comment (maxfierke) [inline spec/fdoc/endpoint_scaffold_spec.rb]: Could this and all the other changed occurrences be `be(true)` instead? `be_truthy` has a slightly different meaning

**Review by maxfierke — COMMENTED** [review]
A couple of small things to address (nothing to spend too much time on), but otherwise looks good

**Review by maxfierke — CHANGES_REQUESTED** [review]
Unfortunately, there's no CI on this fork, but here's what I got from running rspec locally (on both Ruby 3.0 and Ruby 3.2):

<details>
<summary>Failed specs</summary>

<pre>
  1) Fdoc::SpecWatcher on rails should verify when params are a hash
     Failure/Error:
       expect(Fdoc::Service).to receive(:verify!) do |*args|
         expect(args[2]).to eq({ :id => 1 })
       end

       (Fdoc::Service (class)).verify!(*(any args))
           expected: 1 time with any arguments
           received: 0 times with any arguments
     # ./spec/fdoc/spec_watcher_spec.rb:28:in `block (3 levels) in <top (required)>'

  2) Fdoc::SpecWatcher on rails should verify when params are JSON
     Failure/Error: request_params.nil? ? super(*params) : super(action, **request_params)

     TypeError:
       no implicit conversion of String into Hash
     # ./lib/fdoc/spec_watcher.rb:11:in `block (2 levels) in <module:SpecWatcher>'
     # ./spec/fdoc/spec_watcher_spec.rb:38:in `block (3 levels) in <top (required)>'

  3) Fdoc::SpecWatcher on sinatra should verify when params are a hash
     Failure/Error:
       expect(Fdoc::Service).to receive(:verify!)do |*args|
         expect(args[2]).to eq({ :id => 1 })
       end

       (Fdoc::Service (class)).verify!(*(any args))
           expected: 1 time with any arguments
           received: 0 times with any arguments
     # ./spec/fdoc/spec_watcher_spec.rb:64:in `block (3 levels) in <top (required)>'

  4) Fdoc::SpecWatcher on sinatra should verify when params are JSON
     Failure/Error: request_params.nil? ? super(*params) : super(action, **request_params)

     TypeError:
       no implicit conversion of String into Hash
     # ./lib/fdoc/spec_watcher.rb:11:in `block (2 levels) in <module:SpecWatcher>'
     # ./spec/fdoc/spec_watcher_spec.rb:74:in `block (3 levels) in <top (required)>'
</pre>
</details>

**Comment by jjoyner11** [comment]
@maxfierke  I have fixed the `no implicit conversion of String into Hash error`. Other failures are legacy failures that are also coming in master branch.

**Review by maxfierke — APPROVED** [review]
ahh, bummer. well, seems fine for now anyhow

**Pull request merged**